index.js

```diff
@@ -0,0 +1,53 @@
+var SemVer = require('./SemVer');
+var { compareIdentifiers, rcompareIdentifiers } = require('./internal/identifiers');
+var { SEMVER_SPEC_VERSION } = require('./internal/constants');
+var { re, src, t } = require('./internal/re');
```

Comment:
Doing object dereferencing assignment requires a more recent node version than let/const and classes.  May as well update to let/const, classes, and arrow functions in this change as well, and we can make it require node 10 and higher.

Reply:
Alternatively, if you think it's worth keeping the existing node version support for this change, remove all the object dereferencing.

Reply:
I changed:

- var -> const/let (where appropriate)
- SemVer, Range, Comparator -> classes (but I left the helper functions for Range outside the class since they're sorta kinda private
- functions -> arrows _in the callbacks_. I didn't change the named functions in `/functions` or `/ranges` mostly because then I'd have to change it like below in order to keep the names:
```js
module.exports = function compare(...) { ... }
//to
const compare = (...) => {...}
module.exports = compare
```
- string concat -> template strings. I know you didn't request it, but it stuck out like a sore thumb once I had all the other es2015+ stuff in there 😄

Reply:
Oh I also fixed all the casing stuff. 👍